src/shared/cgroup-setup.c

```diff
@@ -384,6 +384,20 @@ int cg_attach(const char *controller, const char *path, pid_t pid) {
         return 0;
 }
 
+int cg_fd_attach(int fd, pid_t pid) {
+        char c[DECIMAL_STR_MAX(pid_t) + 2];
```

Comment:
`DECIMAL_STR_MAX` includes space for NUL, so +2 -> +1?

Reply:
yeah, this is a mess across the whole codebase.

We discuss this every other year. Last time we figured DECIMAL_STR_MAX() should return the value without the NUL, and we and the NUL manually. And I treat it that way ever since, even if it's not implemented that way currently, under the asusmption that that's the safest option: better add too much than too little...

so dunno, it's not clear what to do here

Reply:
Given that `char buf[DECIMAL_STR_MAX(type)]` is used everywhere, I don't think it's necessary... It makes more sense to include the space for NUL IMO.

Reply:
so the reason last time this was discussed we kind came to the conclusion that it should not include space for the NUL is that so often you format strings with multiple values in them, and in that case the space for the NULs just adds up, even though only one is used.

That said, it just wastes one byte per DECIMAL_STR_MAX() call. Which doesn't really matter I guess.

So maybe the right approach is to pretend the extra space for a NUL wasn't included, but actually it is. Thus we are *always* on the safe side, regardless how you fuck things up.

We have other areas where we do something similar. For example, even if you invoke read_full_file() to read a binary file into memory, we'll always implicitly make the allocation one byte larger and append a NUL byte, just as a safety net, and because wasting a byte doesn't matter. Hence maybe the status quo isn't actually that bad, and the way to go: use the macro as if the NUL byte wasn't included, but actually include it in the macro already.